mantidproject/mantid
# Options -R/-E for runSystemTests is now treated as a regex

**Description of work.**

It is often useful to specify multiple tests to either exclude or include when running the system tests. This is now possible by treating the `-R/-E` options as regexes (as the documentation actually stated it would).

I've also added a new `--dry-run/-N` option that just prints the tests that would run to confirm before trying to run.

**To test:**

Testing the exclude option: 

`./systemtest --quiet --dry-run  -E "ILLDirectGeometryReductionTest.IN4|PVPython" > system-tests.txt`

should generate a list without `ILLDirectGeometryReductionTest.IN4` or `PVPythonTest` in it. 

Conversely,

`./systemtest --quiet --dry-run  -R "ILLDirectGeometryReductionTest.IN4|PVPython" > system-tests.txt`

should print only those tests.

*There is no associated issue.*

*This does not require release notes* because **it is an internal change.**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
